crates/bls_utils/src/verification.rs

```diff
@@ -113,26 +132,28 @@ pub fn verify_seed_exchange_commitment(
         )));
     }
 
-    let dest_id = get_index_in_commitments(
+    let dest_id = match get_index_in_commitments(
         verification_hashes,
         &seed_exchange.shared_secret.dst_base_hash,
-    );
-
-    if dest_id.is_err() {
-        return Err(Box::new(VerificationErrors::SlashableError(String::from(
-            format!(
-                "Invalid field seeds_exchange_commitment.shared_secret.dst_id: {} \n",
-                dest_id.unwrap_err()
-            ),
-        ))));
-    }
+    ) {
+        Ok(id) => id,
+        Err(e) => {
+            return Err(Box::new(VerificationErrors::SlashableError(String::from(
+                format!(
+                    "Invalid field seeds_exchange_commitment.shared_secret.dst_base_hash: {} \n",
+                    e
+                ),
+            ))));
+        }
+    };
 
-    let unwraped = dest_id.unwrap() + 1;
-    let test_id = bls_id_from_u32(unwraped);
+    // F(0) is aways reserved for the aggregated key so we need to start from 1
+    let dest_id = dest_id + 1;
+    let test_id = bls_id_from_u32(dest_id);
 
     let mut cfst: Vec<G1Affine> = Vec::new();
     for pubkey in &initial_commitment.verification_vector.pubkeys {
-        cfst.push(G1Affine::from_compressed(pubkey).into_option().unwrap());
+        cfst.push(to_g1_affine(pubkey));
     }
 
     let le_bytes = seed_exchange.shared_secret.dst_id.clone();
```

Comment:
This is about the following `String::from` on line 164, but I cannot leave a comment there, since it's not part of this PR.

For converting &str to String, there are multiple ways, none of which is idiomatic. But you might want to consider using `to_owned()`, as "it conveys the most intent": https://users.rust-lang.org/t/what-is-the-idiomatic-way-to-convert-str-to-string/12160/7